c++ training/W2S1/E2/Source.cpp

```diff
@@ -0,0 +1,62 @@
+#include <iostream>
+
+//Exercise 2 - Military Clock
+//Create a class called MilitaryClock that stores and displays the military time.
+//
+//Your class should contain the following constructors and methods :
+//	MilitaryClock();
+//MilitaryClock(short hours, short minutes);
+//void SetTime(short hours, short minutes);
+//void DisplayTime();
+//
+//Here�s an example of how the class could be used :
+//MilitaryClock clock;
+//clock.Display();
+//clock = MilitaryClock(7, 0);
+//clock.Display();
+//clock.SetTime(12, 15);
+//clock.Display();
+//
+//Don�t forget that military time is displayed as hhmm so add a preceding 0 before hours / minutes if their value is less than 10.
+
+class MilitaryClock {
+private:
+
+	short m_hours, m_minutes;
+
+public:
+
+	MilitaryClock() {
+		m_hours = 0;
+		m_minutes = 0;
+	}
+
+	MilitaryClock(short hours, short minutes) {
+		m_hours = hours;
+		m_minutes = minutes;
+	}
+
+	void SetTime(short hours, short minutes) {
+		m_hours = hours;
+		m_minutes = minutes;
+	}
+
+	void Display() {
+		std::cout << addOptionalZero(m_hours) << m_hours << addOptionalZero(m_minutes) << m_minutes << std::endl;
+	}
+
+	std::string addOptionalZero(short input) {
```

Comment:
Your code makes sense and it's good. However, take a look at this method. It looks descriptive and follows the 'start with a verb' convention. However, it's not as descriptive as it looks. 

It's returning a string, but it doesn't make it clear what the string means. What does add zero mean and is it really optional? 

What are we doing then? We're basically taking a number and converting it to a string. We're also making sure that it is at least two digits. Finally, we're _getting_ a value, we're not actually adding anything to anything,, the function just returns new data. So a much clearer name would be something like `GetAsTwoDigitString()` or similar.

It's clear that we're returning a string based on the number parameter and it's clear that we are formatting to always follow two digits. What do you think?

Reply:
Yup it's confusing... ummm... the thing is.. I was avoiding string concatenation, so what that function actually does is just get a string that is either a 0 or an empty string, that's why it's "optional".  It doesn't actually return the value of the input, it's in fact being added in line 45 after the optional 0 gets printed, so I'm not entirely sure how to proceed here.